### PR TITLE
enable basic indexing for fieldtype data

### PR DESCRIPTION
**Pull request by jxn**
This PR is to allow indexing of the field data.  The keys, values, and ids of the selected options will be stored in solr.

### CODE REVIEW

File: Core/FieldType/EnhancedSelection/SearchField.php [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+        foreach ($fieldDefinition->fieldTypeConstraints->fieldSettings['options'] as $option) {
+            if (in_array($option['identifier'], $selectionKeys)) {
+                $selectionIds[] = $option['id'];
+                $selectionNames[] = $option['name'];
+            }
+        }
+        return [
+            new Search\Field(
+                'enhanced_selection_identifiers',
```
Comment (emodric) [inline Core/FieldType/EnhancedSelection/SearchField.php]: Is it required to prefix the fields with `enhanced_selection`? Field names will be generated from field identifier as it is.

**Comment by emodric** [comment]
Thanks @jxn ! This looks good I think. Besides the inline comment, can you take care of CS to match the current rules? Just run `php-cs-fixer fix` in root of the repo.

**Comment by jxn** [comment]
@emodric : I've run `php-cs-fixer fix` and made those changes.  I haven't tested it since change, but there doesn't appear to be any material difference.  Can you take one last peek?

**Comment by emodric** [comment]
Thanks @jxn !

**Pull request merged**